src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -30,24 +34,111 @@
     private final ScheduledExecutorService executor;
     private final RxJavaSchedulersHook schedulersHook;
     volatile boolean isUnsubscribed;
-
+    /** The purge frequency in milliseconds. */
+    private static final String FREQUENCY_KEY = "io.reactivex.rxjava.scheduler.jdk6.purge-frequency-millis";
```

Comment:
Why can't we fix this then go work on #2579 which still needs to be reviewed and can be rebased?